pkg/scheduler/scheduler.go

```diff
@@ -237,16 +237,29 @@ func (s *Scheduler) FrontendLoop(frontend schedulerpb.SchedulerForFrontend_Front
 
 		switch msg.GetType() {
 		case schedulerpb.ENQUEUE:
-			err = s.enqueueRequest(frontendCtx, frontendAddress, msg)
+			// Extract tracing information from headers in HTTP request. FrontendContext doesn't have the correct tracing
+			// information, since that is a long-running request.
+			tracer := opentracing.GlobalTracer()
+			parentSpanContext, err := httpgrpcutil.GetParentSpanForRequest(tracer, msg.HttpRequest)
+			if err != nil {
+				return err
+			}
```

Comment:
We should not exit `FrontendLoop` due to error in single request. That will effectivelly stop query-scheduler.

Reply:
previously we would send error back to frontend, we can still do that.

Reply:
what about opentracing.NoopTracer{}.StartSpan("noop").Context()